### PR TITLE
Added max idle time

### PR DESCRIPTION
**Pull request by shanejonas**
This adds a timer to check if a plugin has been idling too long. defaults to checking every 5 seconds if any plugins have been idle for 30 seconds.

Also adds `status` to the plugin state, which can be `idle`, `running`, `stopped`, or `crashed`.

### CODE REVIEW

File: packages/controllers/src/plugins/PluginController.test.ts [hunk 1]
```diff
@@ -600,14 +598,16 @@ describe('PluginController Controller', () => {
           code: 123,
         },
       );
-    }, 0);
+    }, 100);
 
     await new Promise((resolve) => {
+      console.log('subbing');
```
Comment (shanejonas) [inline packages/controllers/src/plugins/PluginController.test.ts:604]:
```suggestion
```

File: packages/controllers/src/plugins/PluginController.ts [hunk 2]
```diff
@@ -370,7 +459,7 @@ export class PluginController extends BaseController<
       throw new Error(`Plugin "${pluginName}" not found.`);
     }
 
-    return plugin.isRunning;
+    return plugin.status === 'running';
```
Comment (rekmarks) [inline packages/controllers/src/plugins/PluginController.ts:462]:
```suggestion
    return plugin.status === PluginStatus.running;
```

File: packages/controllers/src/plugins/PluginController.ts [hunk 3]
```diff
@@ -602,7 +691,7 @@ export class PluginController extends BaseController<
   ): Promise<ProcessPluginReturnType> {
     // if the plugin is already installed and active, just return it
     const plugin = this.get(pluginName);
-    if (plugin?.isRunning) {
+    if (plugin?.status !== 'running') {
```
Comment (rekmarks) [inline packages/controllers/src/plugins/PluginController.ts:694]:
```suggestion
    if (plugin?.status !== PluginStatus.running) {
```

**Review by rekmarks — CHANGES_REQUESTED** [review]

**Review by rekmarks — APPROVED** [review]
LGTM!

We have some cleanup to do here in the future that precedes this PR, which includes (1) add more inline documentation, especially docstrings and (2) stop using `any` for controller state in `this.update` calls. Just filing that for later!

**Pull request merged**